api.md

```diff
@@ -0,0 +1,73 @@
+* `/api`
+    * `GET` - Information about the API (version, metadata, links to repo, supported endpoints) - TODO
+
+* `/api/years`
+    * `GET` - List of all supported years. Structure: `{"years": ["2017"]}`
+
+* `/api/years/<year>`
+    * `GET` - Detailed attributes/metadata for a given year. Structure: [years](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/year.json)
+
+* `/api/years/<year>/institutions`
+    * `GET` - Complete list of institutions a user can/could file for a given year. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json)
+
+* `/api/institutions/`
+    * `GET` - Complete list of institutions a user can/could file for *across all years*. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json), separated by year.
+
+* `/api/institutions/<institution>`
+    * `GET` - Data for a single institution *across all years*. Structure: [institution](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json#L48-L56) collected in array based on id (`{institutions: [...]}`).
+
+* `/api/institutions/<institution>/years`
+    * `GET` - List of all years that a given institution filed/will file for. Structure: `{"years": ["2017"]}`
```

Comment:
I think we can remove the `/will file for`. I don't think we would know that information until a new panel is created.